cloudmarker/clouds/azsql.py

```diff
@@ -0,0 +1,231 @@
+                :meth:`_get_server_db_details`.
+
+        """
+        try:
+            tenant = self._tenant
+            creds = self._credentials
+            sub_id = sub.get('subscription_id')
+            sql_client = SqlManagementClient(creds, sub_id)
+            db_server_list = sql_client.servers.list()
+
+            for server_index, sql_server in enumerate(db_server_list):
+                sql_server = sql_server.as_dict()
+
+                _log.info('Found SQL Server #%d: %s; %s',
+                          server_index, sql_server.get('name'),
+                          util.outline_az_sub(sub_index, sub, tenant))
+                sql_server_id = sql_server.get('id')
+                sql_server_name = sql_server.get('name')
+                rg_name = \
+                    tools.parse_resource_id(sql_server_id)['resource_group']
+                yield (sub, rg_name, sql_server_name,
+                       util.outline_az_sub(sub_index, sub, tenant))
+
+                # Break after pulling data for self._max_recs number
+                # of SQL servers for a subscriber. Note that if
+                # self._max_recs is 0 or less, then the following
+                # condition never evaluates to True.
+                if server_index + 1 == self._max_recs:
+                    _log.info('Stopping get_subscription_dbs fetch due '
+                              'to _max_recs: %d; %s', self._max_recs,
+                              util.outline_az_sub(sub_index, sub, tenant))
+                    break
+        except Exception as e:
+            _log.error('Failed to fetch SQL servers; %s; error: %s: %s',
+                       util.outline_az_sub(sub_index, sub, tenant),
+                       type(e).__name__, e)
+
+    def _get_server_db_details(self, sub, rg_name, server_name, sub_summ):
+        """Get details of all DBs for a SQL server.
+
+        Arguments:
+            sub (Subscription): Azure subscription object.
+            rg_name (str): Resource group name.
+            server_name (str): Name of the SQL server.
+            sub_summ (str): Summary of the subscription
+
+        Yields:
+            dict: An Azure SQL server record with TDE configuration.
+
+        """
+        sub_id = sub.get('subscription_id')
+        creds = self._credentials
+        sql_client = SqlManagementClient(creds, sub_id)
+        db_list = sql_client.databases.list_by_server(rg_name, server_name)
+        for db_index, db in enumerate(db_list):
+            sql_db_name = db.as_dict().get('name')
+            tde_config = \
+                sql_client.transparent_data_encryptions.get(rg_name,
+                                                            server_name,
+                                                            sql_db_name)
+            _log.info('Found sql_db #%d: %s: %s; %s',
+                      db_index, sql_db_name, server_name,
+                      sub_summ)
```

Comment:
I suggest writing this log in this format: `'Found sql_db #%d: %s; SQL server #%d: %s; %s'`.

This would make it clear while reading the logs which part is the DB name and which part is the server name.